### PR TITLE
GH1945 Revise code formatting, general copyedit per submitted PR

### PR DESCRIPTION
**Pull request by shrielenee**
## This PR is a:
[ ] New topic
[X ] Content fix or rewrite
[ ] Bug fix or improvement
 

## Summary

Revised code formatting, general copyedit per submitted PR (which requested 2.0 EOL changes): https://github.com/magento/devdocs/pull/1945 

When this pull request is merged, it will fix code sample formatting and provide a general copyedit of three pages in v2.1, which are symlinked to v2.2 and v2.3:
* architecture/archi_perspectives/components/modules/mod_depend_types.md
* extension-dev-guide/build/component-registration.md
* extension-dev-guide/build/create_component.md

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/devdocs?pullRequest=1952) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/devdocs?pullRequest=1952) before we can accept your contribution.<br/>

File: guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md [hunk 1]
```diff
@@ -1,62 +1,50 @@
...
+* Contain code that directly uses logic from another module, as for the latter module's instances, class constants, static methods, public class properties, interfaces, and traits.
+* Contain strings that include class names, method names, class constants, class properties, interfaces, and traits from another module.
+* Deserializes an object declared in another module.
+* Uses or modifies the database tables used by another module.
 
-* The module directly checks another module's availability.
+### Soft dependencies
 
-* The module extends another module's configuration.
+Modules with a *soft dependency* on another module can function properly without the other module, even if they have a dependency upon it. These modules:
```
Comment (bdenham) [inline guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md]: Consider singular version "**A module** with a soft dependency...". Regardless, omit: "even if they have a dependency upon it."

File: guides/v2.1/extension-dev-guide/build/component-registration.md [hunk 2]
```diff
@@ -21,9 +17,9 @@ Register modules with:
 
      ComponentRegistrar::register(ComponentRegistrar::MODULE, '<VendorName_ModuleName>', __DIR__);
 
-where &lt;VendorName> is the name of the company providing the {% glossarytooltip c1e4242b-1f1a-44c3-9d72-1d5b1435e142 %}module{% endglossarytooltip %} and &lt;ModuleName> is the name of the module.
+Here `<VendorName>` is the name of the company providing the {% glossarytooltip c1e4242b-1f1a-44c3-9d72-1d5b1435e142 %}module{% endglossarytooltip %} and `<ModuleName> is the name of the module.
```
Comment (bdenham) [inline guides/v2.1/extension-dev-guide/build/component-registration.md]: add extra tick mark `<ModuleName>**`**

File: guides/v2.1/extension-dev-guide/build/component-registration.md [hunk 3]
```diff
@@ -34,49 +30,44 @@ Register themes with:
 
      ComponentRegistrar::register(ComponentRegistrar::THEME, '<area>/<vendor>/<theme name>', __DIR__);
 
-where &lt;area> is the functional area of the module (frontend, controller, and so on.), &lt;vendor> is the name of the company providing the theme, and &lt;theme name> is the name of the {% glossarytooltip d2093e4a-2b71-48a3-99b7-b32af7158019 %}theme{% endglossarytooltip %}.
+Here `<area>` is the functional area of the module (frontend, controller, and so on.), `<vendor>` is the name of the company providing the theme, and `&lt;theme name>` is the name of the {% glossarytooltip d2093e4a-2b71-48a3-99b7-b32af7158019 %}theme{% endglossarytooltip %}.
```
Comment (bdenham) [inline guides/v2.1/extension-dev-guide/build/component-registration.md]: `<theme name>`

File: guides/v2.1/extension-dev-guide/build/create_component.md [hunk 4]
```diff
@@ -12,48 +9,46 @@ redirect_from:
...
-## Overview of naming a component {#overview-naming}
 You give a name to your component in its `composer.json` and `module.xml` files. These files also contain other required configuration parameters, such as the module's schema version.
 
 ## Prerequisites {#prereq}
 Before you continue, make sure you have completed all of the following tasks:
-
-*   Created a [file structure]({{ page.baseurl }}/extension-dev-guide/build/module-file-structure.html)
-*   Created the the [configuration files]({{ page.baseurl }}/extension-dev-guide/build/required-configuration-files.html) you'll need
-*   [Registered]({{ page.baseurl }}/extension-dev-guide/build/component-registration.html) your component
+.
```
Comment (bdenham) [inline guides/v2.1/extension-dev-guide/build/create_component.md]: remove

File: guides/v2.1/extension-dev-guide/build/create_component.md [hunk 5]
```diff
@@ -12,48 +9,46 @@ redirect_from:
...
+*   Create a [file structure]({{page.baseurl}}/extension-dev-guide/build/module-file-structure.html).
+*   Create the the [configuration files]({{page.baseurl}}/extension-dev-guide/build/required-configuration-files.html) you'll need.
+*   [Register]({{page.baseurl}}/extension-dev-guide/build/component-registration.html) your component.
 
 ## Add the component's `module.xml` file {#module-xml}
-Declare the component itself by adding a {% glossarytooltip c1e4242b-1f1a-44c3-9d72-1d5b1435e142 %}module{% endglossarytooltip %}.xml file in the `/etc` folder of your component.
+Declare the component itself by adding a {% glossarytooltip c1e4242b-1f1a-44c3-9d72-1d5b1435e142 %}`module`{% endglossarytooltip %}.xml file in the `/etc` folder of your component.
```
Comment (bdenham) [inline guides/v2.1/extension-dev-guide/build/create_component.md]: make sure that `module.xml` (including the .xml) is enclosed in the ticks.

File: guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md [hunk 6]
```diff
@@ -1,62 +1,50 @@
...
-Modules with a *hard dependency* on another module cannot function without the module it depends on. Specifically:
-
-* The module contains code that directly uses logic from another module  (for example, the latter module's instances, class constants, static methods, public class properties, interfaces, and traits).
-
-* The module contains strings that include class names, method names, class constants, class properties, interfaces, and traits from another module.
+Magento installs modules in the following order:
 
-* The module deserializes an object declared in another module.
+1. The module serving as a dependency for another module
+2. The module dependent on it
```
[the remark this answers is not in the document]
Comment (shrielenee) [reply guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md]: I'm actually creating a small "Module install order" section. I don't think this content belongs in any of the available sections of this page currently.

File: guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md [hunk 7]
```diff
@@ -1,62 +1,50 @@
...
 
-* The module uses or modifies the database tables used by another module.
+### Hard dependencies
 
-## Soft dependencies
+Modules with a *hard dependency* on another module cannot function without the module it depends on. These modules:
```
[the remark this answers is not in the document]
Comment (shrielenee) [reply guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md]: 👍

File: guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md [hunk 8]
```diff
@@ -1,62 +1,50 @@
...
 
-Modules with a *soft dependency* on another module can function properly without the other module, even if it has a dependency upon it. Specifically:
+* Contain code that directly uses logic from another module, as for the latter module's instances, class constants, static methods, public class properties, interfaces, and traits.
```
[the remark this answers is not in the document]
Comment (shrielenee) [reply guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md]: I believe this extra info is needed.

File: guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md (hunk 1 above)
Comment (shrielenee) [reply guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md]: 👍

File: guides/v2.1/extension-dev-guide/build/component-registration.md (hunk 2 above)
Comment (shrielenee) [reply guides/v2.1/extension-dev-guide/build/component-registration.md]: 👍

**Comment by jfrontain** [comment]
@bdenham, I'll enter comments as soon as I meet my current release note deliverables. Thanks.

File: guides/v2.1/extension-dev-guide/build/component-registration.md (hunk 3 above)
Comment (shrielenee) [reply guides/v2.1/extension-dev-guide/build/component-registration.md]: 👍

File: guides/v2.1/extension-dev-guide/build/create_component.md (hunk 4 above)
Comment (shrielenee) [reply guides/v2.1/extension-dev-guide/build/create_component.md]: 👍

File: guides/v2.1/extension-dev-guide/build/create_component.md (hunk 5 above)
Comment (shrielenee) [reply guides/v2.1/extension-dev-guide/build/create_component.md]: 👍

File: guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (bdenham) [reply guides/v2.1/architecture/archi_perspectives/components/modules/mod_depend_types.md]: Yeah, possibly. So consider changing it to something more direct: "...that directly accesses another module's instance, class constants, static methods..."

**Pull request merged**